lib/apiservers/engine/backends/container.go

```diff
@@ -882,198 +877,174 @@ func IsContainerRunning(containerID string) (bool, error) {
-				inCancel()
-			}
+			err := copyStdErr(ctx, plClient, vc, clStderr)
 			if err != nil {
-				if _, ok := err.(*interaction.ContainerGetStderrNotFound); ok {
-					errors <- derr.NewRequestNotFoundError(fmt.Errorf("No such container: %s", name))
-					return
-				}
-				if _, ok := err.(*interaction.ContainerGetStderrInternalServerError); ok {
-					errors <- derr.NewErrorWithStatusCode(fmt.Errorf("Server error from the interaction port layer: %s", err),
-						http.StatusInternalServerError)
-					return
-				}
-
-				// If we get here, most likely something went wrong with the port layer API server
-				// These errors originate within the go-swagger client itself.
-				// Go-swagger returns untyped errors to us if the error is not one that we define
-				// in the swagger spec.  Even EOF.  Therefore, we must scan the error string (if there
-				// is an error string in the untyped error) for the term EOF.
-				unknownErrMsg := fmt.Errorf("Unknown error from the interaction port layer: %s", err)
-				if strings.Contains(unknownErrMsg.Error(), swaggerSubstringEOF) {
-					log.Info("Detected EOF from swagger, detaching all streams...")
-					inCancel()
-					outCancel()
-				}
-				errors <- derr.NewErrorWithStatusCode(unknownErrMsg, http.StatusInternalServerError)
-				return
+				log.Errorf("container attach: stderr (%s): %s", vc.ContainerID, err.Error())
+			} else {
+				log.Infof("container attach: stderr (%s) done: %s", vc.ContainerID)
 			}
 
-			errors <- nil
+			cancel()
+			errors <- err
 		}()
 	}
 
-	//	go func() {
-	//		select {
-	//			case <- attachContext.Done()
-	//		}
-	//	}()
-
 	// Wait for all stream copy to exit
 	wg.Wait()
-	log.Debugf("Attach stream closed")
-	defer close(errors)
+
+	log.Infof("container attach:  cleaned up connections to %s.", vc.ContainerID)
 	for err := range errors {
 		if err != nil {
+			// If we get here, most likely something went wrong with the port layer API server
+			// These errors originate within the go-swagger client itself.
+			// Go-swagger returns untyped errors to us if the error is not one that we define
+			// in the swagger spec.  Even EOF.  Therefore, we must scan the error string (if there
+			// is an error string in the untyped error) for the term EOF.
+
+			log.Errorf("container attach error: %s", err.Error())
+
 			return err
 		}
 	}
+
+	return nil
+}
+
+func copyStdIn(ctx context.Context, pl *client.PortLayer, vc *viccontainer.VicContainer, clStdin io.ReadCloser, keys []byte) error {
+	// Pipe for stdin so we can interject and watch the input streams for detach keys.
+	stdinReader, stdinWriter := io.Pipe()
+
+	defer stdinWriter.Close()
+
+	go func() {
+		defer stdinReader.Close()
+		// Copy the stdin from the CLI and write to a pipe.  We need to do this so we can
+		// watch the stdin stream for the detach keys.
+		var err error
+		if vc.Config.Tty {
+			_, err = copyEscapable(stdinWriter, clStdin, keys)
+		} else {
+			_, err = io.Copy(stdinWriter, clStdin)
```

Comment:
I preserved previous behavior here but moved around the anon routine.  You're right though;  without `vc.Config.Tty` being set, the pipe/copy is irrelevant.  I'll fix it possibly in a later PR.  Great catch!